src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -545,19 +546,60 @@ private async Task<DurableConfig> ReadDurableTaskConfig()
         // This is a stopgap approach to get the Durable extension version. It duplicates some logic in ExtensionManager.cs.
         private async Task<string> GetDurableMajorVersionAsync(JObject hostJson, ScriptJobHostOptions hostOptions)
         {
+            string metadataFilePath;
             bool isUsingBundles = hostJson != null && hostJson.TryGetValue("extensionBundle", StringComparison.OrdinalIgnoreCase, out _);
-            if (isUsingBundles)
+            // This feature flag controls whether to opt out of the SyncTrigger metadata fix for OOProc DF apps from https://github.com/Azure/azure-functions-host/pull/9331
+            if (FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagEnableLegacyDurableVersionCheck))
```

Comment:
The new behavior will be on by default. Will this break existing apps that are currently running and potentially working around the existing behavior via config?

Reply:
To clarify: this bug fix is motivated by ScV3 which does not have this compensating logic. That's how we discovered this bug